script.js

```diff
@@ -1,4 +1,739 @@
+            playersArray[i].totalPoints - playersArray[i].currentRoundWager;
+        }
+        if (computerScore == playersArray[i].score) {
+          playersArray[i].result = 'It is a draw';
+        }
+      }
+      // User gets blackjack and computer does not
+      if (playersArray[i].score == 21 && computerScore != 21) {
+        playersArray[i].result = 'You get blackjack.<br>You win';
+        playersArray[i].totalPoints =
+          playersArray[i].totalPoints + playersArray[i].currentRoundWager;
+      }
+      // Computer gets blackjack and user does not
+      if (computerScore == 21 && playersArray[i].score != 21) {
+        playersArray[i].result = 'Computer gets blackjack.<br>Computer wins';
+        playersArray[i].totalPoints =
+          playersArray[i].totalPoints - playersArray[i].currentRoundWager;
+      }
+      // Both user and computer gets blackjack
+      if (computerScore == 21 && playersArray[i].score == 21) {
+        playersArray[i].result = 'You get blackjack.<br>It is a draw';
+      }
+      // User goes over and computer does not
+      if (playersArray[i].score > 21 && computerScore <= 21) {
+        playersArray[i].result = 'You have gone over.<br>Computer wins';
+        playersArray[i].totalPoints =
+          playersArray[i].totalPoints - playersArray[i].currentRoundWager;
+      }
+      // Computer goes over and user does not
+      if (computerScore > 21 && playersArray[i].score < 21) {
+        playersArray[i].result = 'You win';
+        playersArray[i].totalPoints =
+          playersArray[i].totalPoints + playersArray[i].currentRoundWager;
+      }
+      // Both user and computer goes over
+      if (playersArray[i].score > 21 && computerScore > 21) {
+        playersArray[i].result = 'You have gone over.<br>It is a draw';
+      }
+    }
+    // Create the scoreboard and return the scoreboard
+    scoreboard = `Dealer's cards: ${computerCards}<br>Dealer's score: ${computerScore}<br><br>`;
+    for (i = 0; i < playersArray.length; i += 1) {
+      console.log(playersArray[i].score);
+      console.log(playersArray[i].result);
+      scoreboard =
+        scoreboard +
+        `${playersArray[i].name}'s cards: ${playersArray[i].output}<br>${playersArray[i].name}'s score: ${playersArray[i].score}<br>${playersArray[i].result}<br>${playersArray[i].name}'s total points: ${playersArray[i].totalPoints}<br><br>`;
+    }
+
+    //Restart game variables
+    for (i = 0; i < playersArray.length; i += 1) {
+      playersArray[i].score = 0;
+      playersArray[i].currentRoundWager = 0;
+      playersArray[i].cards = [];
+      playersArray[i].decision = '';
+      playersArray[i].seen = false;
+    }
+    computerCardsArray = [];
+    computerResult = '';
+    gameStatus = 'wager';
+    firstRound = 1;
+    computerDecisionOutput = '';
+    return (
+      scoreboard +
+      `<br><br>${playersArray[0].name}, key in your wager for the next round and click submit`
+    );
```

Comment:
Logic looks sound :) 

A quick note. If you find that the code in your main function is really long. It's likely an indication that you could split it up into helper functions.